contracts/ThalesRoyale/ThalesRoyaleVoucher.sol

```diff
@@ -0,0 +1,124 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-4.4.1/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-4.4.1/utils/Counters.sol";
+import "@openzeppelin/contracts-4.4.1/access/Ownable.sol";
+import "@openzeppelin/contracts-4.4.1/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts-4.4.1/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts-4.4.1/token/ERC20/utils/SafeERC20.sol";
+
+contract ThalesRoyaleVoucher is ERC721URIStorage, Ownable {
+    /* ========== LIBRARIES ========== */
+
+    using Counters for Counters.Counter;
+    using SafeMath for uint;
+    using SafeERC20 for IERC20;
+
+    /* ========== STATE VARIABLES ========== */
+
+    Counters.Counter private _tokenIds;
+
+    string public _name = "Thales Royale Voucher";
+    string public _symbol = "TRV";
+    bool public paused = false;
+    string public tokenURI;
+
+    address public thalesRoyaleAddress;
+
+    IERC20 public sUSD;
+    uint public price;
+    mapping(uint => uint) public pricePerVoucher;
```

Comment:
NFTs are supposed to be stateful, cant this be a part of the minted NFT storage?